mle/model.py

```diff
@@ -3,110 +3,72 @@
+    def stream(self, chat_history: List[Dict[str, str]], **kwargs) -> str:
         pass
 
-
 class OllamaModel(Model):
-    def __init__(self, model: str = 'llama3', host_url=None):
-        """
-        Initialize the Ollama model.
-        Args:
-            host_url (str): The Ollama Host url.
-            model (str): The model version.
-        """
+    def __init__(self, model: str = 'llama3', host_url: Optional[str] = None):
         super().__init__()
+        self.model = model
+        self.model_type = MODEL_OLLAMA
+        self.ollama = self._import_ollama()
+        self.client = self.ollama.Client(host=host_url)
 
+    def _import_ollama(self):
         dependency = "ollama"
         spec = importlib.util.find_spec(dependency)
-        if spec is not None:
-            self.model = model
-            self.model_type = MODEL_OLLAMA
-            self.ollama = importlib.import_module(dependency)
-            self.client = self.ollama.Client(host=host_url)
-        else:
+        if spec is None:
             raise ImportError(
-                "It seems you didn't install ollama. In order to enable the Ollama client related features, "
-                "please make sure ollama Python package has been installed. "
-                "More information, please refer to: https://github.com/ollama/ollama-python"
+                "Ollama Python package is not installed. "
+                "Please install it to use Ollama-related features. "
+                "More information: https://github.com/ollama/ollama-python"
             )
+        return importlib.import_module(dependency)
 
-    def query(self, chat_history, **kwargs):
-        """
-        Query the LLM model.
-        Args:
-            chat_history: The context (chat history).
-        """
+    def query(self, chat_history: List[Dict[str, str]], **kwargs) -> str:
         return self.client.chat(model=self.model, messages=chat_history)['message']['content']
 
-    def stream(self, chat_history, **kwargs):
-        """
-        Stream the output from the LLM model.
-        Args:
-            chat_history: The context (chat history).
-        """
-        for chunk in self.client.chat(
-                model=self.model,
-                messages=chat_history,
-                stream=True
-        ):
+    def stream(self, chat_history: List[Dict[str, str]], **kwargs) -> str:
+        for chunk in self.client.chat(model=self.model, messages=chat_history, stream=True):
             yield chunk['message']['content']
 
-
 class OpenAIModel(Model):
-    def __init__(self, api_key, model='gpt-3.5-turbo', temperature=0.7):
-        """
-        Initialize the OpenAI model.
-        Args:
-            api_key (str): The OpenAI API key.
-            model (str): The model with version.
-            temperature (float): The temperature value.
-        """
+    def __init__(self, api_key: str, model: str = 'gpt-3.5-turbo', temperature: float = 0.7):
```

Comment:
Traceback (most recent call last):
  File "/root/miniconda3/envs/mle/bin/mle", line 33, in <module>
    sys.exit(load_entry_point('mle-agent', 'console_scripts', 'mle')())
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/root/workspace/MLE-agent/mle/cli.py", line 68, in start
    return baseline(os.getcwd(), model)
  File "/root/workspace/MLE-agent/mle/workflow/baseline.py", line 67, in baseline
    advisor_report = advisor.interact("[green]User Requirement:[/green] " + ml_requirement + "\n" + ask_data(dataset))
  File "/root/workspace/MLE-agent/mle/agents/advisor.py", line 131, in interact
    self.report = self.suggest(requirement)
  File "/root/workspace/MLE-agent/mle/agents/advisor.py", line 113, in suggest
    text = self.model.query(
  File "/root/workspace/MLE-agent/mle/model.py", line 85, in query
    result = get_function(function_name)(**arguments)
NameError: name 'get_function' is not defined